### PR TITLE
Improve Scrollbar drag behavior

### LINKED ISSUE #107765: ScrollBar moves with an offset if cursor leaves flutter application.
This issue is reproducible on both the stable and master branch, and affects both desktop and web applications. I checked the behaviour with web apps such as YouTube and the chrome browser and it seems that they take into consideration the position of the cursor when deciding as to whether to move the ScrollBar.

<details>

<summary>video of bug in Flutter:</summary>

https://user-images.githubusercontent.com/73116038/179329223-c0bf6738-3cfa-4f8f-bcdc-6a88b5e9e6bb.mp4

</details>

<details>

<summary>video of expected behaviour in YouTube:</summary>

https://user-images.githubusercontent.com/73116038/179329228-da40bd35-8cb1-43fb-80b3-fe756663769d.mp4

</details>

<details>

<summary>Reproducible code:</summary>

    import 'package:flutter/material.dart';
    
    void main() => runApp(const MyApp());
    
    class MyApp extends StatelessWidget {
      const MyApp({Key? key}) : super(key: key);
    
      static const String _title = 'Flutter Code Sample';
    
      @override
      Widget build(BuildContext context) {
        return MaterialApp(
          title: _title,
          home: Scaffold(
            appBar: AppBar(title: const Text(_title)),
            body: const Center(
              child: MyStatefulWidget(),
            ),
          ),
        );
      }
    }
    
    class MyStatefulWidget extends StatefulWidget {
      const MyStatefulWidget({Key? key}) : super(key: key);
    
      @override
      State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
    }
    
    class _MyStatefulWidgetState extends State<MyStatefulWidget> {
      final ScrollController _firstController = ScrollController();
    
      @override
      Widget build(BuildContext context) {
        return Scrollbar(
          thumbVisibility: true,
          controller: _firstController,
          child: ListView.builder(
              controller: _firstController,
              itemCount: 100,
              itemBuilder: (BuildContext context, int index) {
                return Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: Text('Scrollable 1 : Index $index'),
                );
              }),
        );
      }
    }

</details>

<details>

<summary>flutter doctor -v</summary>

    [✓] Flutter (Channel master, 3.1.0-0.0.pre.1633, on LMDE 5 (elsie)
        5.10.0-16-amd64, locale en_GB.UTF-8)
        • Flutter version 3.1.0-0.0.pre.1633 on channel master at
          /home/henryr/Dev/SDK/Flutter/flutter
        • Upstream repository https://github.com/flutter/flutter.git
        • Framework revision 95a5a8d741 (2 days ago), 2022-07-13 09:53:40 -0700
        • Engine revision 20038216ac
        • Dart version 2.18.0 (build 2.18.0-272.0.dev)
        • DevTools version 2.15.0
    
    [✓] Android toolchain - develop for Android devices (Android SDK version
        32.1.0-rc1)
        • Android SDK at /home/henryr/Android/Sdk
        • Platform android-32, build-tools 32.1.0-rc1
        • Java binary at:
          /home/henryr/Dev/SDK/AndroidStudio/android-studio/jre/bin/java
        • Java version OpenJDK Runtime Environment (build
          11.0.12+0-b1504.28-7817840)
        • All Android licenses accepted.
    
    [✓] Chrome - develop for the web
        • Chrome at google-chrome
    
    [✓] Linux toolchain - develop for Linux desktop
        • Debian clang version 11.0.1-2
        • cmake version 3.18.4
        • ninja version 1.10.1
        • pkg-config version 0.29.2
    
    [✓] Android Studio (version 2021.2)
        • Android Studio at /home/henryr/Dev/SDK/AndroidStudio/android-studio
        • Flutter plugin version 69.0.2
        • Dart plugin version 212.5744
        • Java version OpenJDK Runtime Environment (build
          11.0.12+0-b1504.28-7817840)
    
    [✓] Connected device (2 available)
        • Linux (desktop) • linux  • linux-x64      • LMDE 5 (elsie) 5.10.0-16-amd64
        • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.114
    
    [✓] HTTP Host Availability
        • All required HTTP hosts are available
    
    • No issues found!

</details>

### LINKED ISSUE #112072: `ScrollBar` moves slower than the mouse when dragging it to scroll
This one is bit different from https://github.com/flutter/flutter/issues/111968. It happens with current main in cases where `SliverList` items extents keep increasing.

Main reason for this is that `ScrollBar._updateScrollPosition` adjusts viewport position by using relative scroll offset. In pathological cases like this the error keeps increasing to the point where not even overdrag is enough to reach end.

I'm wondering why ScrollBar is written like that. Setting `pixels` to `minExtent + (absoluteScrollPosition * (maxExtent - minExtent))` would alleviate this (`absoluteScrollPosition` being in range of 0.0-1.0). But I'm not sure how this would work with overdrag.

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(
    title: 'Flutter Demo',
    home: MyHomePage(),
  ));
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CustomScrollView(
        slivers: [
          SliverToBoxAdapter(
            child: Container(
              height: 2,
              color: Colors.red,
            ),
          ),
          SliverList(
            delegate: SliverChildBuilderDelegate(
              (context, index) => Padding(
                padding: const EdgeInsets.all(10),
                child: Container(
                  color: Colors.green,
                  height: index.toDouble(),
                ),
              ),
              childCount: 1000,
            ),
          ),
          SliverToBoxAdapter(
            child: Container(
              height: 2,
              color: Colors.red,
            ),
          ),
        ],
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by xu-baolin**
This change lets the thumb always follow the pointer even if the scroll extent changed when dragging the thumb.
 
Fixes https://github.com/flutter/flutter/issues/112072
Fixes https://github.com/flutter/flutter/issues/107765

### CODE REVIEW

**Comment by TahaTesser** [comment]
This also fixes https://github.com/flutter/flutter/issues/107765, my [fix](https://github.com/flutter/flutter/pull/111250) for it was reverted. 

Since this seems to be a better fix, I'm assigning https://github.com/flutter/flutter/issues/107765  to you.

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -1446,15 +1457,15 @@ class RawScrollbar extends StatefulWidget {
 /// Provides defaults gestures for dragging the scrollbar thumb and tapping on the
 /// scrollbar track.
 class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProviderStateMixin<T> {
-  Offset? _dragScrollbarAxisOffset;
+  Offset? _startDragScrollbarAxisOffset;
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1460]: @TahaTesser Thanks for the sync info, I tested it and it did fix the issue by accident. : )

**Review by Piinks — COMMENTED** [review]
Thanks for the PR @xu-baolin! I will check out a copy of this tomorrow and test it out! :)

**Comment by xu-baolin** [comment]
> Thanks for the PR @xu-baolin! I will check out a copy of this tomorrow and test it out! :)

Thank you, this maybe also help to implement the infinite scrollbar. You know that the infinite scrollable's scroll extent always changes.

File: packages/flutter/lib/src/widgets/scrollbar.dart (hunk 1 above)
Comment (xu-baolin) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1460]: @Piinks Hi, do you have a chance to take a look at this change?  : )

Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1460]: Ah yes, apologies! I was out of town and this got buried in my inbox. I checked it out before I left and forgot to click submit - sorry! 😓

**Review by Piinks — APPROVED** [review]
LGTM-cat

**Pull request merged**